Maple2.Model/Validators/ClubNameValidator.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Text.RegularExpressions;
+using Maple2.Model.Error;
+using Maple2.Model.Metadata;
+
+namespace Maple2.Model.Validators;
+
+public static partial class ClubNameValidator {
+    private static readonly Regex ValidNamePatternAscii = NamePatternAsciiRegex();
+    private static readonly Regex ValidNamePatternUnicode = NamePatternUnicodeRegex();
+
+    /// <summary>
+    /// Validates a club name according to all rules.
+    /// </summary>
+    /// <param name="name">The club name to validate</param>
+    /// <returns>ClubError code if invalid, null if valid</returns>
+    public static ClubError? ValidateName(string name) {
```

Comment:
im not understanding why we have different validators for club/guild. they dont have varying errors - it's only one error type.

i also don't think we should be returning null. it should just be "ok" or "none". Same goes for ChracterName.

Reply:
In case the name sizes are different or want to be made different